src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,408 @@
+      have diff_eval : differentiable_on ℝ (λ x, (f.map (algebra_map ℤ ℝ)).eval x) (Ioo α (a/b)),
+      { refine differentiable.differentiable_on _, exact (map (algebra_map ℤ ℝ) f).differentiable },
+      rcases (exists_deriv_eq_slope (λ x, (f.map (algebra_map ℤ ℝ)).eval x)
+                α_lt cont_eval diff_eval) with ⟨x₀, x₀_range, hx₀⟩,
+      clear cont_eval diff_eval,
+      simp only [polynomial.deriv, α_root, sub_zero] at hx₀,
+
+      replace hx₀ := hx₀.symm,
+      have Df_x₀_nonzero : (f.map (algebra_map ℤ ℝ)).derivative.eval x₀ ≠ 0 := hx₀.symm ▸ λ rid, hab₂'
+        begin rwa [rid, div_eq_iff (show (a / b : ℝ) - α ≠ 0, by linarith), zero_mul] at hx₀ end,
+
+      replace hx₀ : (↑a / ↑b - α) = (f.map (algebra_map ℤ ℝ)).eval (a/b) / (f.map (algebra_map ℤ ℝ)).derivative.eval x₀,
+      { symmetry, rw div_eq_iff at hx₀ ⊢,  rwa mul_comm,
+        exact Df_x₀_nonzero,
+        rw sub_ne_zero, exact hab₁.symm },
+
+    repeat { have ineq :=
+      calc B / 2 / b^f.nat_degree
+        ≥ abs (α - ↑a / ↑b) : ge_iff_le.2 abs_lt
+    ... = abs((f.map (algebra_map ℤ ℝ)).eval (a/b) / ((f.map (algebra_map ℤ ℝ)).derivative.eval x₀))
+        : by rw [congr_arg abs hx₀, neg_div, abs_neg] <|>
+            rw [show abs (α - a / b) = abs (a / b - α), from abs_sub α (a/b), congr_arg abs hx₀]
+    ... = abs ((f.map (algebra_map ℤ ℝ)).eval (a/b)) / abs ((f.map (algebra_map ℤ ℝ)).derivative.eval x₀)
+        : by rw abs_div
+    ... ≥ 1/b^f.nat_degree / abs ((f.map (algebra_map ℤ ℝ)).derivative.eval x₀)
+        : begin
+            rw ge_iff_le,
+            refine div_le_div (abs_nonneg _)
+              (abs_eval_rat_ge_one_div_denom_pow_n f a b b_pos hab₂')
+              (abs_pos_of_ne_zero Df_x₀_nonzero) (le_refl _),
+          end
+    ... ≥ 1/b^f.nat_degree / M
+        : begin
+            rw ge_iff_le,
+            refine div_le_div _ (le_refl _) (abs_pos_of_ne_zero Df_x₀_nonzero) _,
+            { rw one_div_nonneg, apply pow_nonneg, norm_cast, exact le_of_lt b_pos },
+            { refine hM x₀ _, simp only [mem_Ioo, mem_Icc] at x₀_range hab₀ ⊢, split,
+              { exact le_trans hab₀.1 (le_of_lt x₀_range.1) <|>
+                exact le_trans (by linarith) (le_of_lt x₀_range.1) },
+              { exact le_trans (le_of_lt x₀_range.2) (by linarith) } }
+          end
+    ... = 1/M / b^f.nat_degree
+        : by ring
+    ... ≥ B / b^f.nat_degree
+        : begin
+            rw ge_iff_le,
+            refine div_le_div _ _ _ (le_refl _),
+            { rw one_div_nonneg, exact le_of_lt M_pos },
+            { refine finset.min'_le distances' (1/M) (mem_insert_self _ _) },
+            { apply pow_pos, norm_cast, exact b_pos },
+          end
+    ... > B / 2 / b^f.nat_degree
+        : begin
+            rw [gt_iff_lt, div_lt_div_right],
+            { exact half_lt_self B_pos },
+            { apply pow_pos, norm_cast, exact b_pos },
+          end,
+
+    linarith only [ineq] } }
+end
```

Comment:
This proof is still extremely long.
Also, please format according to the style guide.